thirdparty/github.go

```diff
@@ -1765,6 +1765,81 @@ func GetMergeablePullRequest(ctx context.Context, issue int, githubToken, owner,
 	return pr, nil
 }
 
+// HasEvergreenBranchProtections checks if any branch protection rule begins "evergreen".
+// We deliberately check for the prefix because users may have installed variant-level branch protections.
```

Comment:
To verify, is variant-level branch protection enough? For the CQ results then would we only consider the outcome of those variants or does the whole cq patch have to be successful? (apologies if i've asked you this already haha)

Reply:
We're not checking the branch protection rules in order to decide whether or not to merge a PR. GitHub does that. We're checking to make sure that the user has set at least one branch protection rule that begins "evergreen". Users would either want the whole build to finish, so they would use "evergreen", or they would only want a variant (Cloud wants this), in which case they would use "evergreen/<variant>".